discovery-provider/ddl/pg_migrate.sh

```diff
@@ -0,0 +1,94 @@
+#!/bin/bash
+#
+# PostgreSQL schema migration manager
+# https://github.com/zenwalker/pg_migrate.sh/tree/v2.0
+
+set -e
+[[ -f .env ]] && source .env
+
+echo " ... starting pg_migrate "
+
+MIGRATIONS_DIR="migrations"
+MIGRATIONS_TABLE="schema_version"
+
+POSTGRES_USER=${POSTGRES_USER:-postgres}
+POSTGRES_HOST=${POSTGRES_HOST:-127.0.0.1}
+POSTGRES_PORT=${POSTGRES_PORT:-5432}
+export PGPASSWORD="$POSTGRES_PASSWORD"
+
+# setting DB_URL will override any individual settings from above
+DB_URL=${DB_URL:-postgres://$POSTGRES_USER:$POSTGRES_PASSWORD@$POSTGRES_HOST:$POSTGRES_PORT/$POSTGRES_DB}
+
+# remove +psycopg2 from DB_URL
+DB_URL="${DB_URL/\+psycopg2/}"
+
+alias psql='psql -qtAX -v ON_ERROR_STOP=1 $DB_URL'
+shopt -s expand_aliases
+
+
+create_migrations_table() {
+    migrations_table_exists=$(psql -c "SELECT to_regclass('$MIGRATIONS_TABLE');")
+
+    if  [[ ! $migrations_table_exists ]]; then
+        echo "Creating $MIGRATIONS_TABLE table"
+        psql -c "CREATE TABLE $MIGRATIONS_TABLE (file_name text PRIMARY KEY, md5 text, applied_at TIMESTAMPTZ NOT NULL DEFAULT NOW());"
+    fi
+}
+
+
+migrate_dir() {
+    migration_files=$(ls $1/*.sql | sort -V)
+
+    md5s=$(psql -c "select md5 from $MIGRATIONS_TABLE")
+
+    for file in $migration_files; do
+        md5=$(cat $file | tr -d "[:space:]" | md5sum | awk '{print $1}')
+
+        if [[ $md5s =~ $md5 ]]; then
+          echo "... skipping $file $md5"
+          continue
+        fi
+
+        if [[ $file =~ "failable" ]]; then
+            echo "Applying failable $file"
+            set +e
+            psql < "$file"
+            retval=$?
+            if [[ $retval -ne 0 ]]; then
+              echo "Failable: $file failed: $retval"
+            fi
+            set -e
+        else
+            echo "Applying $file"
+            psql < "$file"
+        fi
+        psql -c "INSERT INTO $MIGRATIONS_TABLE (file_name, md5) VALUES ('$file', '$md5') on conflict(file_name) do update set md5='$md5', applied_at=now();"
+    done
+}
+
+migrate() {
+    create_migrations_table
+
+    migrate_dir $MIGRATIONS_DIR
+    migrate_dir "functions"
+
+    # run non_test files if PG_MIGRATE_TEST_MODE is unset
+    if [[ $PG_MIGRATE_TEST_MODE != "true" ]]; then
+        migrate_dir "non_test"
```

Comment:
want to make sure this doesn't run on prod / stage because i think it would brick migrations with is_current true. also 0x0 doesn't even exist on stage / prod.

Reply:
The intention is to run the `0001_initial_block.sql` in all environments _except_ for test.  It is required for starting the server.  

If you don't create this initial block row in a fresh db the server will be like "expected SINGLE row to be marked current" or whatever.

It's basically copied from [here](https://github.com/AudiusProject/audius-protocol/blob/main/discovery-provider/alembic/versions/2571ec7593c7_init_db_models.py#L26-L41)

Reply:
It is safe because:
* the pkey is `"blocks_pkey" PRIMARY KEY, btree (blockhash)`
* it has on conflict do nothing
* is_current can be any value... the primary key will prevent any subsequent insert